src/measure_theory/measure_space.lean

```diff
@@ -1323,23 +1351,105 @@ def absolutely_continuous (μ ν : measure α) : Prop :=
 
 infix ` ≪ `:50 := absolutely_continuous
 
-lemma absolutely_continuous.mk (h : ∀ ⦃s : set α⦄, measurable_set s → ν s = 0 → μ s = 0) : μ ≪ ν :=
+lemma absolutely_continuous_of_le (h : μ ≤ ν) : μ ≪ ν :=
+λ s hs, nonpos_iff_eq_zero.1 $ hs ▸ le_iff'.1 h s
+
+alias absolutely_continuous_of_le ← has_le.le.absolutely_continuous
+
+lemma absolutely_continuous_of_eq (h : μ = ν) : μ ≪ ν :=
+h.le.absolutely_continuous
+
+alias absolutely_continuous_of_eq ← eq.absolutely_continuous
+
+namespace absolutely_continuous
+
+lemma mk (h : ∀ ⦃s : set α⦄, measurable_set s → ν s = 0 → μ s = 0) : μ ≪ ν :=
 begin
   intros s hs,
   rcases exists_measurable_superset_of_null hs with ⟨t, h1t, h2t, h3t⟩,
   exact measure_mono_null h1t (h h2t h3t),
 end
 
-@[refl] lemma absolutely_continuous.refl (μ : measure α) : μ ≪ μ := λ s hs, hs
+@[refl] protected lemma refl (μ : measure α) : μ ≪ μ := rfl.absolutely_continuous
 
-lemma absolutely_continuous.rfl : μ ≪ μ := λ s hs, hs
+protected lemma rfl : μ ≪ μ := λ s hs, hs
 
-lemma absolutely_continuous_of_eq (h : μ = ν) : μ ≪ ν := by rw h
+@[trans] protected lemma trans (h1 : μ₁ ≪ μ₂) (h2 : μ₂ ≪ μ₃) : μ₁ ≪ μ₃ :=
+λ s hs, h1 $ h2 hs
 
-alias absolutely_continuous_of_eq ← eq.absolutely_continuous
+@[mono] protected lemma map (h : μ ≪ ν) (f : α → β) : map f μ ≪ map f ν :=
+if hf : measurable f then absolutely_continuous.mk $ λ s hs, by simpa [hf, hs] using @h _
+else by simp only [map_of_not_measurable hf]
 
-@[trans] lemma absolutely_continuous.trans (h1 : μ₁ ≪ μ₂) (h2 : μ₂ ≪ μ₃) : μ₁ ≪ μ₃ :=
-λ s hs, h1 $ h2 hs
+end absolutely_continuous
+
+lemma ae_le_iff_absolutely_continuous : μ.ae ≤ ν.ae ↔ μ ≪ ν :=
+⟨λ h s, by { rw [measure_zero_iff_ae_nmem, measure_zero_iff_ae_nmem], exact λ hs, h hs },
+  λ h s hs, h hs⟩
+
+alias ae_le_iff_absolutely_continuous ↔ has_le.le.absolutely_continuous_of_ae
+  measure_theory.measure.absolutely_continuous.ae_le
+alias absolutely_continuous.ae_le ← ae_mono'
+
+lemma absolutely_continuous.ae_eq (h : μ ≪ ν) {f g : α → δ} (h' : f =ᵐ[ν] g) : f =ᵐ[μ] g :=
+h.ae_le h'
+
+/-! ### Quasi measure preserving maps -/
+
+/-- A map `f : α → β` is said to be *quasi measure preserving* w.r.t. measures `μa` and `μb`
```

Comment:
The name I have always seen in the literature for this kind of map is "non-singular". I agree that `quasi_measure_preserving` is better for mathlib, as it is less ambiguous, but you should mention nonsingularity in the docstring to make it more easy to find.

Reply:
I copied the name from your Isabelle file. I can use any name.

Reply:
As I said, I think this is a good name (I am at least coherent with myself :-). But I think one should also mention nonsingular in the docstring.